tests/testthat/setup-testing_depth.R

```diff
@@ -0,0 +1,55 @@
+#' Returns testing depth set by an environmental variable.
+#'
+#' @details Looks for the option `TESTING_DEPTH` first, if not set,
+#' takes the system environmental variable `TESTING_DEPTH`. If neither
+#' is set, then returns 3 by default. If the value of `TESTING_DEPTH`
+#' is not a scalar numeric, then returns 3.
+#'
+#' @return `numeric(1)` the testing depth.
+#'
+get_testing_depth <- function() {
+  default_depth <- 3
+  depth <- getOption("TESTING_DEPTH", Sys.getenv("TESTING_DEPTH", default_depth))
+
+
+  depth <- tryCatch(
+    as.numeric(depth),
+    error = function(error) default_depth,
+    warning = function(warning) default_depth
+  )
+
+  if (length(depth) != 1 || is.na(depth)) {
```

Comment:
Now I think there is no way this is `NA`
```suggestion
  if (length(depth) != 1) {
```

Reply:
True, it will never be `NA` anymore.